### ISSUE TITLE
Function 'CudnnBatchNormBackward' returned nan values in its 0th output

### ISSUE THREAD

**Issue by saniazahan** [kenziyuliu/MS-G3D#40]
Hi thank you so much for sharing your work. I am trying to recreate the results. I am using the ntu xsub dataset you provided with half precision amp level 1. But at epoch 33 I got NaN out from batchnorm layer. I originated from "out = tempconv(x)" function in ms_tcn.py file. I had autograd_anomally on. All the config settings are kept as your repo. Could you please suggest why this happened.

![a](https://user-images.githubusercontent.com/12523634/120953658-06472800-c780-11eb-9246-0f2af63f01cb.png)

**Comment by saniazahan** [comment]
Update: I removed autograd_anomally detection. Its training smoothly for now. Maybe training in half precision is the culprit as autograd detects the gradients overflow as error before GradScaler comes in action.

**Comment by MagicFrogSJTU** [comment]
> Update: I removed autograd_anomally detection. Its training smoothly for now. Maybe training in half precision is the culprit as autograd detects the gradients overflow as error before GradScaler comes in action.

Dude you really save my life

**Comment by Youth-yang** [comment]
Thanks

**Comment by YChienHung** [comment]
amazing solution ！！！
